docs/MULTIREPO.md

```diff
@@ -0,0 +1,99 @@
+# This doc describes a workflow for using multiple repositories in a development flow.
+
+## Problem
+For some of the assets we (will) use in the app we do not have the rights to freely distribute them. Githubs' ToS specify that anything in the public repo must be freely redistributable.
+This means that we need a flow where we have a public repo for doing the main work and a private for storing those assets and putting them in at build time.
+
+## Scope
+This document is only about the mobile app(s), it is not about anything server-side.
+
+## Proposed solution
+We will have 2 repositories set up: App and Overlay.
+
+The App repository will contain everything to build the app on a developers machine.
+The App repository will contain an empty theme directory. When looking for an 
+The Overlay repository will contain:
+  - A submodule, which is the App repository
+  - An overlay directory
+  - Build scripts that will ensure the build uses files from the overlay, if they exist and falls back to defaults otherwise
+
+The overlay directory can only contain white-listed paths; this prevents the Overlay repository from accidentally adding features. 
+Furthermore the overlay directory can only contain files that are present in the App repository; this prevents situations where the App repository doesn't build without the Overlay repository.
+Any images in the the Overlay repo must have the exact same dimensions as the corresponding image in the App repo.
+
+Identified paths for the overlay:
+  - client/flutter/assets
+  - client/flutter/android/app/src/main/res
+  - client/flutter/ios/Runner/Assets.xcassets
+
+From a developers' perspective there will be 2 scenarios: you're working on the app, or you're working on the design.
+If you work on the app you only use the App repository.
+If you're working on design you will need to checkout the Overlay repository, including the App repo submodule.
+
+Process 1: create a new public asset
+- Work in the App repo, add the asset as usual
+
+Process 2: replace a public asset with a private one
+- Create your private asset in the overlay directory in the same path of the public asset you want to replace.
+
+Process 3: create a new private asset
+- Do process 1
+- Do process 2
+
+Process 4: remove a private asset
+- Remove the file from the overlay folder
+
+Process 5: remove a public asset
+- Do process 4
+- Delete the asset from the App repo
+
+
+While this process will work for unique assets, there is an issue with scaled assets. Suppose we have 3 variants of an app icon that also have private versions.
+If a 4th variant gets added in the App repo, we want to make sure that the build in the Overlay repo fails until we've added that variant as well.
```

Comment:
Why not just go through Process 3 for the medium-size asset?

It is correct that there is no way for a build script to detect that we're actually missing a specific version of a private asset. However, this concern also applies for private assets in general -- if someone adds an entire new asset to the public repo, there's no way for a build script to detect that we're missing its private version. I think in either case we have to rely on PR reviewers.

Reply:
Note that the default will be that we don't need a private version of each public asset. This is different for a new variant of a public asset that has a private version.

We can go ahead and start with a manual process though. In the future we could trigger a ci fail on new public assets, until they are explicitly set to not have a private version.

Reply:
Got it. What if we use a naming convention for public assets that also have private versions? such as `name_private.png` -- this way, we will make explicit when we have a private version for a public asset (so it will be clear when alternate versions are added that the private counterpart will be needed, too).

This way, we can also use a CI script to enforce that all file names matching the `_private` pattern should have a corresponding file on the private repo.

Reply:
I don't want any changes to the public repo in favor of the private one, if at all possible.